aspnetcore/fundamentals/routing/samples/3.x/RoutingSample/Controllers/ProductController.cs

```diff
@@ -9,11 +9,11 @@ namespace RoutingSample.Controllers
     public class ProductController : Controller
     {
         public IActionResult Index() =>
-            ControllerContext.ToActionResult();
+            ControllerContext.MyDisplayRouteInfo();
```

Comment:
I still find myself really unsure about why this is here. Does the documentation actually mention anything about this?

Reply:
> Does the documentation actually mention anything about this?

Yes, the following

The [MyDisplayRouteInfo](https://github.com/dotnet/AspNetCore.Docs/tree/master/aspnetcore/fundamentals/routing/samples/3.x/RoutingSample/Extensions/ControllerContextExtensions.cs) method is included in the [sample download](https://github.com/dotnet/AspNetCore.Docs/tree/master/aspnetcore/fundamentals/routing/samples/3.x) and is used to display routing information.

> I still find myself really unsure about why this is here. 

Customers won't put up with code that doesn't compile. All the snippets in the routing docs work and are testable. Customers love being able to copy/paste/test/modify/test code. I'll keep a close watch on this but I'm sure customers will love the ability to run the sample.

Reply:
I'm just not sure why something like this is part of our docs. The code could compile by returning an empty ContentResult.

Reply:
But that's not testable by customers. They like to test the code and verify it's working. If every route returned an empty content result, how could you tell which route was run?

It's a big part of the sample download. Even without downloading the entire repo, customers can copy/paste the code they're interested and test them out.

Routes are one of the biggest problems for our customers and this gives them a way to do quick testing and experimenting.

Why don't we see how customers react?

Reply:
If that's the desire can you make this into a Razor view? (something like what you showed me before).

Reply:
In the areas test code, I [return a string](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/mvc/controllers/areas/31samples/MVCareas/Extensions/ControllerContextExtensions.cs#L15-L18)
I use [ViewData to save the string](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/mvc/controllers/areas/31samples/MVCareas/Areas/Products/Controllers/HomeController.cs#L10)

A partial or layout displays `ViewData["routeInfo"]`

That works great for MVC with controllers and view and Razor pages. For Web API's, I could do something like:

```
[Route("api/[controller]")]
[ApiController]
public class TestController : ControllerBase
{
    [HttpGet]
    public IActionResult ListProducts()
    {
        return Content(CC.MyDisplayRouteInfo());
    }
}
```

I don't remember what i showed you before.

Reply:
For consistency, why don't I make them all 

```
public IActionResult ListProducts()
    {
        return Content(CC.MyDisplayRouteInfo());
    }
```